doc/cephfs/nfs.rst

```diff
@@ -79,3 +79,234 @@ Current limitations
+
+By default replicated size for OSD is 3. Since we are using only one OSD. It can cause error. Let's fix this up by setting replicated size to 1.::
+
+        [root@minikube /]# ceph osd pool set cephfs.myfs.meta size 1
+        [root@minikube /]# ceph osd pool set cephfs.myfs.data size 1
+
+.. note:: The replicated size should never be less than 3 in real production deployment.
+
+Check Cluster status again::
+
+        [root@minikube /]# ceph -s
+          cluster:
+            id:     3a30f44c-a9ce-4c26-9f25-cc6fd23128d0
+            health: HEALTH_OK
+
+          services:
+            mon: 3 daemons, quorum a,b,c (age 27m)
+            mgr: a(active, since 27m)
+            mds: myfs:1 {0=myfs-a=up:active} 1 up:standby-replay
+            osd: 1 osds: 1 up (since 56m), 1 in (since 56m)
+
+          data:
+            pools:   2 pools, 24 pgs
+            objects: 22 objects, 2.2 KiB
+            usage:   5.1 GiB used, 11 GiB / 16 GiB avail
+            pgs:     24 active+clean
+
+          io:
+            client:   639 B/s rd, 1 op/s rd, 0 op/s wr
+
+Create a NFS-Ganesha Server Cluster
+-----------------------------------
+Add Storage for NFS-Ganesha Servers to prevent recovery conflicts::
+
+        [root@minikube /]# ceph osd pool create nfs-ganesha 64
+        pool 'nfs-ganesha' created
+        [root@minikube /]# ceph osd pool set nfs-ganesha size 1
+        [root@minikube /]# ceph orchestrator nfs add mynfs nfs-ganesha ganesha
+
+Here we have created a NFS-Ganesha cluster called "mynfs" in "ganesha"
+namespace with "nfs-ganesha" OSD pool.
+
+Scale out NFS-Ganesha cluster::
+
+        [root@minikube /]# ceph orchestrator nfs update mynfs 2
+
+Configure NFS-Ganesha Exports
+-----------------------------
+Initially rook creates ClusterIP service for the dashboard. With this service
+type, only the pods in same kubernetes cluster can access it.
+
+Expose Ceph Dashboard port::
+
+        kubectl patch service -n rook-ceph -p '{"spec":{"type": "NodePort"}}' rook-ceph-mgr-dashboard
+        kubectl get service -n rook-ceph rook-ceph-mgr-dashboard
+        NAME                      TYPE       CLUSTER-IP       EXTERNAL-IP   PORT(S)          AGE
+        rook-ceph-mgr-dashboard   NodePort   10.108.183.148   <none>        8443:31727/TCP   117m
+
+This makes the dashboard reachable outside kubernetes cluster and the service
+type is changed to NodePort service.
+
+Create JSON file for dashboard::
+
+        $ cat ~/export.json
+        {
+              "cluster_id": "mynfs",
+              "path": "/",
+              "fsal": {"name": "CEPH", "user_id":"admin", "fs_name": "myfs", "sec_label_xattr": null},
+              "pseudo": "/cephfs",
+              "tag": null,
+              "access_type": "RW",
+              "squash": "no_root_squash",
+              "protocols": [4],
+              "transports": ["TCP"],
+              "security_label": true,
+              "daemons": ["mynfs.a", "mynfs.b"],
+              "clients": []
+        }
+
```

Comment:
Do we really want to enshrine this procedure for the future?

The blog post I did was mainly there to show people that the backend infrastructure to do this is now there, but the intent was always to eventually wire this into the dashboard for a more reasonable UI. As we've already found, these interfaces and procedures are somewhat prone to bitrot.

Reply:
After we complete the ceph fs volume work for creating the exports. I will update the doc.

Reply:
After we complete the ceph fs volume work for creating the exports. I will update the doc.